### ISSUE TITLE
vision data export script

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#1282]
Create a script to compile a CSV file containing photos with the following columns:

* `photo_id`
* `set` (training, validation, or testing)
* `species_id` (taxon ID for species, ID of parent species for infraspecies) 
* `taxon_id`
* `observation_id`
* `medium_image_url`

Requirements
1. Only use species observed by 20 or more people with at lest 8 photos from RG observations (the Target Species)
1. No user can contribute to more than one set **per species**. Users should be randomly assigned to **per species** sets
1. Photos should be chosen randomly within a unique set and species
1. Maintain a 2:1:2 ratio of training:validation:test **per species**, with minimum counts of 8 training, 4 validation, and 8 testing photos **per species**
1. Test photos must come from RG observations, validation and training photos from RG or Needs ID observations
1. Export list of all species names and IDs from the Target Species
1. Add an `augmented_test` set to the photos that includes the same number of photos as the `test` set but from RG observations of species outside of the Target Species. Doesn't matter how many photos per species.

My first stab at interpreting Scott's procedure from https://docs.google.com/document/d/1cj6O1GxciiLGYAhj--6nZJ5ByjJlSUI871SaCxUY_I8/edit saved for posterity:

Procedure for selecting these data:

1. **Generate species categories:** Create a category for each species (with ssp/var rolled up into parent species) with at least 20 distinct observation_photos from 20 distinct users from RG observations. We’ll call this the species_id
1. **Designate train/val/test users:** For each species_id, no user should contribute photos to more than one of the Train, Val, Test sets. Divide all users into Test or Train or Val. While assignment to these groups should be random, it’s ok if they always get assigned to the same groups (i.e. user_id % 3 should be fine).
1. **Select the observation_photos:** For each category, select observation_photos for the following sets:
    * **Test:** Using only photos from the Test users, select observation_photos from RG obs such that the number of observation photos represent 40% of the total available from Research Grade observations.
    * **Train:** Using only photos from the Train users, select all observation_photos from Verifiable obs
    * **Val:** Using only photos from the Val users, select all observation_photos from Verifiable obs
1. **Augment Test dataset with observation_photos from taxa outside the categories:** Randomly select a set of observation_photos from RG observations of species not represented by the categories, i.e. species with less than 20 RG observations by unique users. The size of the set should be equal to the size of the test set represented by the categories, i.e. 40% of the total number of photos in RG observations.

**Comment by alexshepard** [comment]
I would just do a training / testing split. The validation segment is specific to neural networks.

**Comment by kueda** [comment]
So just 50 / 50 training / testing?

**Comment by loarie** [comment]
I guess your right that you could do 40% Test and 60% Train+Val but you'll have to add user_id so you can create Train and Val groups with distinct users after the fact

**Comment by kueda** [comment]
So... what do you guys want, a three-way split or a two-way with user_id?
